nnunetv2/training/dataloading/test_dataloader.py

```diff
@@ -0,0 +1,160 @@
+
+        os.environ['MASTER_ADDR'] = 'localhost'
+        if 'MASTER_PORT' not in os.environ.keys():
+            port = str(find_free_network_port())
+            print(f"using port {port}")
+            os.environ['MASTER_PORT'] = port  # str(port)
+
+        mp.spawn(run_ddp_dataloading,
+                 args=(
+                     dataset_name_or_id,                     
+                     configuration,
+                     fold,
+                     trainer_class_name,
+                     plans_identifier,
+                     use_compressed_data,
+                     disable_checkpointing,
+                     continue_training,
+                     only_run_validation,
+                     pretrained_weights,
+                     export_validation_probabilities,
+                     val_with_best,
+                     num_gpus),
+                 nprocs=num_gpus,
+                 join=True)
+    else:
+        
+        nnunet_trainer = get_trainer_from_args(dataset_name_or_id, configuration, fold, trainer_class_name,
+                                            plans_identifier, use_compressed_data, device=device)
+
+        
+        if torch.cuda.is_available():
+            cudnn.deterministic = False
+            cudnn.benchmark = True
+
+        nnunet_trainer.run_dataloading_test()
+
+
+def run_dataloader_test_entry():
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument('dataset_name_or_id', type=str,
+                        help="Dataset name or ID to train with")
+    parser.add_argument('configuration', type=str,
+                        help="Configuration that should be trained")
+    parser.add_argument('fold', type=str,
+                        help='Fold of the 5-fold cross-validation. Should be an int between 0 and 4.')
+    parser.add_argument('-tr', type=str, required=False, default='nnUNetTrainer',
+                        help='[OPTIONAL] Use this flag to specify a custom trainer. Default: nnUNetTrainer. To run the Pytorch DataLoader, use "nnUNetTrainerPyTorchDataloader"')
+    parser.add_argument('-p', type=str, required=False, default='nnUNetPlans',
+                        help='[OPTIONAL] Use this flag to specify a custom plans identifier. Default: nnUNetPlans')
+    parser.add_argument('-num_gpus', type=int, default=1, required=False,
+                        help='Specify the number of GPUs to use for training')
+    parser.add_argument("--use_compressed", default=False, action="store_true", required=False,
+                        help="[OPTIONAL] If you set this flag the training cases will not be decompressed. Reading compressed "
+                             "data is much more CPU and (potentially) RAM intensive and should only be used if you "
+                             "know what you are doing")
+    parser.add_argument('-device', type=str, default='cuda', required=False,
```

Comment:
It's more typical to use -- for options and - for flags

Reply:
hmm ok will change. This part of the code is picked up as is from nnUNet.